src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -208,3 +214,50 @@ object RetrieveByExternalIdsWithIgnoreUnknownIds {
       value => value.items
     )
 }
+
+trait RetrieveByInstanceIds[R, F[_]] extends WithRequestSession[F] with BaseUrl {
+  def retrieveByInstanceIds(instanceIds: Seq[InstanceId]): F[Seq[R]]
+  def retrieveByInstanceId(instanceId: InstanceId): F[R] =
+    requestSession.map(
+      retrieveByInstanceIds(Seq(instanceId)),
+      (r1: Seq[R]) =>
+        r1.headOption.getOrElse(
+          throw new SdkException("Unexpected empty response when retrieving item by InstanceId")
+        )
+    )
+}
+
+object RetrieveByInstanceIds {
+  def retrieveByInstanceIds[F[_], R](
+      requestSession: RequestSession[F],
+      baseUrl: Uri,
+      instanceIds: Seq[CogniteIdOrInstance]
```

Comment:
```suggestion
      instanceIds: Seq[InstanceId]
```
why not just instanceId, externalId object only handles externalId for example

Reply:
some places do id || externalId in same path, some split but class, not entirely sure if we currently need instance or externalId or internalId case :thinking: will need to read more to get more context..

Reply:
It's missing encoder for that to work but actually can add it. Don't think this makes much difference.

Reply:
Actually no, there is a distinction: Seq[InstanceId] is encoded as [{"space": "space", "externalId": "extId"}] whereas Seq[CogniteIdOrInstance] is encoded as [{"instanceId": {"space": "space", "externalId": "extId"}}] and the first one is incorrect.

Reply:
how it's encoded may need to be handled, but still not sure if retrieveByInstanceId should accept CogniteId as input option